src/index.ts

```diff
@@ -37,10 +41,157 @@ function gistIdFromTwiddleUrl(twiddleUrl: string): string {
   }
 }
 
-export default function twiddleEject(twiddleUrl: string, projectName: string, _options: {} = {}) {
+/**
+ * Given a path, generate single or multi level directory.
+ * @param path
+ */
+function generateDir(path: string) {
+  const fullPath = path.split('/').reduce((accumulator, current) => {
+    if (!fs.existsSync(accumulator)) {
+      fs.mkdirSync(accumulator);
+    }
+    return `${accumulator}/${current}`;
+  });
+  if (!fs.existsSync(fullPath)) {
+    fs.mkdirSync(fullPath);
+  }
+}
+
+/**
+ * Unzip and extract from a given url and save the data to a path provided.
+ * @param gistUrl
+ * @param filePath
+ * @param targetDir
+ */
+async function unzipAndExtractFromUrl(gistUrl: string, filePath: string, targetDir: string): Promise<string> {
+  const gistLink = await fetch(gistUrl).then(res => res.url);
+  const gistDataUrl = `${gistLink}/${filePath}`;
+  const newZip = new AdmZip();
+  const gistData = await fetch(gistDataUrl)
+    .then(x => x.arrayBuffer())
+    .then(x => Buffer.from(x));
+  const zip = new AdmZip(gistData);
+  const zipEntries = zip.getEntries();
+
+  zipEntries.forEach(zipEntry => {
+    const fileName = zipEntry.entryName;
+    const fileContent = zip.readAsText(fileName);
+    // Here remove the top level directory
+    const newFileName = fileName.substring(fileName.indexOf('/') + 1);
+
+    // tslint:disable-next-line: no-bitwise
+    newZip.addFile(newFileName, (fileContent as unknown) as Buffer, '', 0o644 << 16);
+  });
+
+  // Generate the project root directory.
+  generateDir(targetDir);
+  newZip.extractAllTo(targetDir);
+  // tslint:disable-next-line: no-console
+  console.log(`Files extracted to path: ${targetDir}`);
```

Comment:
Let's use the [`debug`](https://www.npmjs.com/package/debug) npm package for this

```ts
import * as debug from 'debug';

const log = debug('twiddle-eject');

log(`Files extracted to path: ${targetDir}`);
```

And you'd be able to see it you did something like this in the CLI to set the `DEBUG` env variable appropriately

```sh
DEBUG=twiddle-eject* ember-twiddle-eject <url>
```